test/test-pmctl-local.js

```diff
@@ -3,20 +3,127 @@ var childctl = require('strong-control-channel/process');
 var fs = require('fs');
-var helper = require('./helper');
 var path = require('path');
+var tap = require('tap');
 var util = require('util');
 
-require('shelljs/global');
+var pm;
+var port;
 
-process.env.STRONGLOOP_CLUSTER = 0;
+function setup(t) {
+  var helper = require('./helper');
+  delete helper.ok; // Use tap to determine ok
 
-console.log('Done setup, run process manager');
+  process.env.STRONGLOOP_CLUSTER = 0;
 
-function onReceive() {
+  t.test('setup', {timeout: 10000}, function(t) {
+    if (port) {
+      t.assert(true, 'reuse pm setup');
+      return t.end();
+    }
+    _pm = manager(function(_port) {
+      var pmurl = util.format('http://127.0.0.1:%d/default', _port);
+      console.log('pmurl: %s', pmurl);
+
+      cp.exec(util.format('git push %s master:master', pmurl), function(er) {
+        assert.ifError(er);
+        port = _port;
+        pm = _pm;
+        t.assert(true, 'pm started on port ' + _port);
+        t.end();
+      });
+    });
+  });
+
+  waiton(t, 'status', /current:$/m);
+}
+
+function expect(t, cmd, pattern) {
+  var name = 'expect: ' + cmd + ' ' + (pattern || '');
+  t.test(name, function(t) {
+    console.log("START %s", name);
+    pmctl(cmd, function(out) {
+      console.log("check %s with pattern %j against code: %j",
+        name, pattern, out.code);
+
+      t.equal(out.code, 0, 'pmctl exit code');
+
+      if (out.code == 0)
+        t.assert(checkOutput(out, pattern), pattern || '(no pattern)');
+
+      if (out.code != 0 || !checkOutput(out, pattern))
+        console.log('check failed against: <\n%>', out.output);
+      t.end();
+    });
+  });
+}
+
+function waiton(t, cmd, pattern) {
+  var name = 'waiton: ' + cmd + ' ' + (pattern || '');
+  t.test(name, {timeout: 10000}, function(t) {
+    console.log("START %s", name);
+    check();
+
+    function check() {
+      pmctl(cmd, function(out) {
+        console.log("check %s against code: %j", name, out.code);
+        if (out.code == 0 && checkOutput(out, pattern)) {
+          t.equal(out.code, 0, 'pmctl exit code');
+          t.assert(true, pattern || '(no pattern)');
+          t.end();
+          return;
+        }
+        setTimeout(check, 1000);
+      });
+    }
+  });
+}
+
+function failon(t, cmd, pattern) {
+  var name = 'failon: ' + cmd + ' ' + (pattern || '');
+  t.test(name, function(t) {
+    console.log("START %s", name);
+    pmctl(cmd, function(out) {
+      console.log("check %s against code: %j", name, out.code);
+
+      t.notEqual(out.code, 0);
+      // FIXME(sam) output assertions were missing from the original, I'll
+      // add them back in once I know the original test passes, because
+      // the patterns may not be correct
+      // t.assert(checkOutput(out, pattern), pattern || '(no pattern)');
+      t.end();
+    });
+  });
+}
+
+function pmctl(cmd, callback) {
+  var cli = require.resolve('../bin/sl-pmctl.js');
+  cmd = cli + ' ' + cmd;
+  return cp.exec(cmd, function(er, stdout, stderr) {
```

Comment:
I'm pretty sure if Windows does manage to exec the file, it won't do it right (will likely use Windows scripting host).

Should use `cp.execFile(process.execPath, [ cli, cmd.split(' ') ], ...` or something like that.

Gotchas include:
 - Windows will exec a .js file with Windows Scripting Host, which will fail..
 - `process.execPath` may have spaces in it, so Array form is better